presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/WindowMatcher.java

```diff
@@ -46,19 +72,153 @@ public MatchResult detailMatches(PlanNode node, PlanNodeCost cost, Session sessi
         return toStringHelper(this)
-                .add("specification", specification)
+                .omitNullValues()
+                .add("prePartitionedInputs", prePartitionedInputs.orElse(null))
+                .add("specification", specification.orElse(null))
+                .add("preSortedOrderPrefix", preSortedOrderPrefix.orElse(null))
+                .add("hashSymbol", hashSymbol.orElse(null))
                 .toString();
     }
+
+    public static class Builder
+    {
+        private final PlanMatchPattern source;
+        private Optional<Set<SymbolAlias>> prePartitionedInputs = Optional.empty();
+        private Optional<ExpectedValueProvider<WindowNode.Specification>> specification = Optional.empty();
+        private Optional<Integer> preSortedOrderPrefix = Optional.empty();
+        private List<AliasMatcher> windowFunctionMatchers = new LinkedList<>();
+        private Optional<Optional<SymbolAlias>> hashSymbol = Optional.empty();
+
+        Builder(PlanMatchPattern source)
+        {
+            this.source = requireNonNull(source, "source is null");
+        }
+
+        public Builder prePartitionedInputs(Set<String> prePartitionedInputs)
+        {
+            requireNonNull(prePartitionedInputs, "prePartitionedInputs is null");
+            this.prePartitionedInputs = Optional.of(
+                    prePartitionedInputs.stream()
+                            .map(SymbolAlias::new)
+                            .collect(toImmutableSet()));
+            return this;
+        }
+
+        public Builder specification(
+                List<String> partitionBy,
+                List<String> orderBy,
+                Map<String, SortOrder> orderings)
+        {
+            return specification(PlanMatchPattern.specification(partitionBy, orderBy, orderings));
+        }
+
+        public Builder specification(ExpectedValueProvider<WindowNode.Specification> specification)
+        {
+            this.specification = Optional.of(specification);
+            return this;
+        }
+
+        public Builder preSortedOrderPrefix(int preSortedOrderPrefix)
+        {
+            this.preSortedOrderPrefix = Optional.of(preSortedOrderPrefix);
+            return this;
+        }
+
+        public Builder addFunction(String output, ExpectedValueProvider<FunctionCall> functionCall)
+        {
+            return addFunction(Optional.of(output), functionCall);
+        }
+
+        public Builder addFunction(ExpectedValueProvider<FunctionCall> functionCall)
+        {
+            return addFunction(Optional.empty(), functionCall);
+        }
+
+        private Builder addFunction(Optional<String> output, ExpectedValueProvider<FunctionCall> functionCall)
+        {
+            windowFunctionMatchers.add(new AliasMatcher(output, new WindowFunctionMatcher(functionCall, Optional.empty(), Optional.empty())));
+            return this;
+        }
+
+        public Builder addFunction(
+                String output,
```

Comment:
I was hoping to indicate what the string was an alias of -- namely, the symbol bound to the result of the function.  I'll change it here and elsewhere to "outputAlias".